lib/rules/no-extra-parens.js

```diff
@@ -348,25 +348,21 @@ module.exports = function(context) {
             }
         },
         "ExpressionStatement": function(node) {
-            var firstToken;
+            var firstToken, secondToken, firstTokens;
             if (hasExcessParens(node.expression)) {
-                firstToken = context.getFirstToken(node.expression);
-
-                // Pure object literals ({}) do not need parentheses but
-                // member expressions do ({}.toString())
-                if ((
-                        firstToken.value !== "{" ||
-                        node.expression.type === "ObjectExpression"
-                    ) &&
-                    // For such as `(function(){}.foo.bar)`
+                firstTokens = context.getFirstTokens(node.expression, 2);
+                firstToken = firstTokens[0];
+                secondToken = firstTokens[1];
+
+                if (
+                    !firstToken ||
```

Comment:
I don't love using a falsiness test here, but eslint yelled at me for using `== null`, and I'm not about to try to figure out whether I should check for `undefined` or `null` values in the array returned by this function.